src/main/java/org/jstepanovic/server/service/Server.java

```diff
@@ -6,50 +6,64 @@
 import java.net.InetAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
+import java.net.SocketException;
+import java.util.concurrent.*;
 
 import static org.jstepanovic.commons.Constants.SERVER_ADDRESS;
 import static org.jstepanovic.commons.Constants.SERVER_PORT;
 
 
-public class Server {
+public enum Server {
+    INSTANCE;
 
-    private final int numberOfProcessors;
-
-    private static Server instance;
+    private final int numberOfProcessors = Runtime.getRuntime().availableProcessors();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(numberOfProcessors);
 
     private final Repository repository = new Repository();
 
-    private Server() {
-        numberOfProcessors = Runtime.getRuntime().availableProcessors();
-    }
+    private boolean isServerRunning;
```

Comment:
If you will try to read this variable in different threads you may see some awkward results. You may read about happens-before
Caution, this one is pretty long and mind-blowing:)
So maybe you want to find something else on this topic
[Java Memory Model Pragmatics](https://shipilev.net/blog/2014/jmm-pragmatics/)

Reply:
using AtomicBoolean